src/atoms/__tests__/referencesState.update.test.ts

```diff
@@ -1,11 +1,20 @@
 import { createStore } from 'jotai';
 
-import { updateProjectReference } from '../../api/referencesAPI';
+import * as referencesAPI from '../../api/referencesAPI';
 import { REFERENCES } from '../../features/references/__tests__/test-fixtures';
 import { ReferenceItem } from '../../types/ReferenceItem';
 import { getDerivedReferenceAtom, setReferencesAtom, updateReferenceAtom } from '../referencesState';
 
-vi.mock('../../api/referencesAPI');
+vi.mock('../../api/referencesAPI', async () => {
+  const actual: object = await vi.importActual('../../api/referencesAPI');
```

Comment:
Mocking the whole module also mocks UPDATABLE_FIELDS which is now also used by ReferencesState for testable functions. I used importActual to mock only the functions that need to be mocked to avoid overwriting UPDATABLE_FIELDS. When using importActual, checking the mocked function directly results in this error: "[Function updateProjectReference] is not a spy or a call to a spy!" I don't know why. Updated to use importOriginal as first argument rather than importActual direct.

Reply:
Ok! That makes sense.

About the error you got, you can write
```typescript
{
    ...actual,
    updateProjectReference: vi.fn(),
}
```
so that the function is a mocked function instead of a regular function, and so you can check it in tests. But I don't mind the spy version